compiler/AST/UseStmt.cpp

```diff
@@ -403,6 +399,62 @@ void UseStmt::writeListPredicate(FILE* mFP) const {
   }
 }
 
+/************************************* | **************************************
+*                                                                             *
+* Determine if the provided type was named in some form in the import         *
+* statement.  Used for determining if we should traverse the import statement *
+* to find its methods.                                                        *
+*                                                                             *
+************************************** | *************************************/
+std::set<const char*> UseStmt::typeWasNamed(Type* t) const {
+  std::set<const char*> namedTypes;
+
+  typeWasNamed(t, &namedTypes);
+  return namedTypes;
+}
+
+void UseStmt::typeWasNamed(Type* t, std::set<const char*>* namedTypes) const {
+  if (isPlainUse() == true) {
+    // We don't limit the use in any way, so we don't need special handling to
+    // find methods
+    return;
+
+  } else {
+    std::string nameToCheck = t->symbol->name;
+    if (AggregateType* at = toAggregateType(t)) {
+      if (at->instantiatedFrom != NULL) {
+        // Need to check against the generic type's name rather than the
+        // instantiation, since the instantiation name included instantiation
+        // information in it (and that isn't usable in a use/import list, at
+        // least right now)
+        AggregateType* rootType = at->getRootInstantiation();
+        nameToCheck = rootType->symbol->name;
+      }
+    }
+    const char* name = astr(nameToCheck.c_str());
```

Comment:
see my comments on the Import version about astrs - no need for `std::string` or any `astr` calls at all here